In [3]:
#IMPORTS

import os
import json
import geopandas as gpd
import numpy as numpy
import pandas as pd
import seaborn as sns

# Setting up barrios_gdf for cleaning

In this section, we read barrios_carcteristicas.geojson from datos/datos_out and we transform it into a dataframe using geopandas library

In [4]:
#SETTING UP barrios_caracteristicas as GDF

# SETTING UP BARRIOS DF
path = '../datos/datos_out/barrios_caracteristicas.geojson'


#CARGAMOS LOS DATOS DE LOS BARRIOS
with open(path) as json_file:
    json_data = json.load(json_file)

barrios_json = []
for i in range(len(json_data['features'])):
    barrios_json.append(json_data['features'][i])           #Guardamos resto de campos de geojson


barrios_gdf = gpd.GeoDataFrame.from_features(barrios_json)
barrios_gdf.crs = 'epsg:4326' #Aseguramos que la proyección es la adecuada para coordenadas GPS

barrios_gdf

geometry coddistrit  \
0    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
1    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
2    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
3    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
4    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
..                                                 ...        ...   
115  POLYGON ((-0.39047 39.47825, -0.38996 39.47754...          3   
116  POLYGON ((-0.37567 39.47381, -0.37423 39.47351...          1   
117  POLYGON ((-0.35018 39.42223, -0.35011 39.42227...         19   
118  POLYGON ((-0.36720 39.50425, -0.36767 39.50338...         15   
119  POLYGON ((-0.35839 39.45904, -0.35801 39.45867...         10   

     gis_gis_barrios_area  object_id_barrio  linkid codbarrio coddistbar  \
0             917112.5625                62       0         5        115   
1             917112.5625                62       0         5        115   
2             917112.5625                62       0         5        115   
3             917112.5625                62       0         5        115   
4             917112.5625                62       0         5        115   
..                    ...               ...     ...       ...        ...   
115           496668.7500                28       0         3        033   
116           438658.3125                37       0         6        016   
117                   NaN               249       0         3        193   
118                   NaN               201       0         1        151   
119           540119.6250                57       0         5        105   

                                   geo_point_2d  nombre_barrio  \
0      [39.44628964870906, -0.3326600366971329]       NATZARET   
1      [39.44628964870906, -0.3326600366971329]       NATZARET   
2      [39.44628964870906, -0.3326600366971329]       NATZARET   
3      [39.44628964870906, -0.3326600366971329]       NATZARET   
4      [39.44628964870906, -0.3326600366971329]       NATZARET   
..                                          ...            ...   
115  [39.473565513491664, -0.39098170775459257]     LA PETXINA   
116    [39.47010643870186, -0.3760489821791233]  SANT FRANCESC   
117  [39.406290326215334, -0.33967854582733203]         PINEDO   
118   [39.494312122911424, -0.3699310988383476]    ELS ORRIOLS   
119    [39.45287103221434, -0.3659376655648462]     NA ROVELLA   

    last_edited_user last_edited_date  %_zona_verde  nivel_acustico  \
0               None             None      0.118869             3.0   
1               None             None      0.118869             3.0   
2               None             None      0.118869             3.0   
3               None             None      0.118869             3.0   
4               None             None      0.118869             3.0   
..               ...              ...           ...             ...   
115             None             None      0.087284             4.0   
116             None             None      0.134529             4.0   
117             None             None           NaN             3.0   
118             None             None      0.153242             3.0   
119             None             None      0.216731             3.0   

     num_hospitales  num_colegios  num_chargestations     calidad_ambiental  \
0               1.0           6.0                 NaN  Razonablemente Buena   
1               1.0           6.0                 NaN  Razonablemente Buena   
2               1.0           6.0                 NaN  Razonablemente Buena   
3               1.0           6.0                 NaN  Razonablemente Buena   
4               1.0           6.0                 NaN  Razonablemente Buena   
..              ...           ...                 ...                   ...   
115             1.0          11.0                 NaN                  None   
116             NaN       

In [6]:
# CHECKING HOW MANY NULL VALUES ARE IN MY DF
barrios_gdf.isnull().sum()

geometry                  0
coddistrit                0
gis_gis_barrios_area     54
object_id_barrio          0
linkid                    0
codbarrio                 0
coddistbar                0
geo_point_2d              0
nombre_barrio             0
last_edited_user        119
last_edited_date        119
%_zona_verde             27
nivel_acustico            0
num_hospitales           62
num_colegios             18
num_chargestations      100
calidad_ambiental        78
num_contenedores          6
num_papeleras             6
num_transporte            8
dtype: int64

# Dropping columns we don't want
Attending to last_edited_user and last_edited_date we see that there are 119 missing values, which are for the whole column of the dataframe.
We are going to drop them using .drop() method

In [7]:
# DROPPING FULL NULL COLUMNS

barrios_gdf = barrios_gdf.drop(['last_edited_user', 'last_edited_date'], axis = 1)
barrios_gdf

geometry coddistrit  \
0    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
1    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
2    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
3    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
4    POLYGON ((-0.33459 39.45478, -0.33326 39.45487...         11   
..                                                 ...        ...   
115  POLYGON ((-0.39047 39.47825, -0.38996 39.47754...          3   
116  POLYGON ((-0.37567 39.47381, -0.37423 39.47351...          1   
117  POLYGON ((-0.35018 39.42223, -0.35011 39.42227...         19   
118  POLYGON ((-0.36720 39.50425, -0.36767 39.50338...         15   
119  POLYGON ((-0.35839 39.45904, -0.35801 39.45867...         10   

     gis_gis_barrios_area  object_id_barrio  linkid codbarrio coddistbar  \
0             917112.5625                62       0         5        115   
1             917112.5625                62       0         5        115   
2             917112.5625                62       0         5        115   
3             917112.5625                62       0         5        115   
4             917112.5625                62       0         5        115   
..                    ...               ...     ...       ...        ...   
115           496668.7500                28       0         3        033   
116           438658.3125                37       0         6        016   
117                   NaN               249       0         3        193   
118                   NaN               201       0         1        151   
119           540119.6250                57       0         5        105   

                                   geo_point_2d  nombre_barrio  %_zona_verde  \
0      [39.44628964870906, -0.3326600366971329]       NATZARET      0.118869   
1      [39.44628964870906, -0.3326600366971329]       NATZARET      0.118869   
2      [39.44628964870906, -0.3326600366971329]       NATZARET      0.118869   
3      [39.44628964870906, -0.3326600366971329]       NATZARET      0.118869   
4      [39.44628964870906, -0.3326600366971329]       NATZARET      0.118869   
..                                          ...            ...           ...   
115  [39.473565513491664, -0.39098170775459257]     LA PETXINA      0.087284   
116    [39.47010643870186, -0.3760489821791233]  SANT FRANCESC      0.134529   
117  [39.406290326215334, -0.33967854582733203]         PINEDO           NaN   
118   [39.494312122911424, -0.3699310988383476]    ELS ORRIOLS      0.153242   
119    [39.45287103221434, -0.3659376655648462]     NA ROVELLA      0.216731   

     nivel_acustico  num_hospitales  num_colegios  num_chargestations  \
0               3.0             1.0           6.0                 NaN   
1               3.0             1.0           6.0                 NaN   
2               3.0             1.0           6.0                 NaN   
3               3.0             1.0           6.0                 NaN   
4               3.0             1.0           6.0                 NaN   
..              ...             ...           ...                 ...   
115             4.0             1.0          11.0                 NaN   
116             4.0             NaN           2.0                 2.0   
117             3.0             1.0           2.0                 NaN   
118             3.0             1.0           9.0                 NaN   
119             3.0             1.0           7.0                 NaN   

        calidad_ambiental  num_contenedores  num_papeleras  num_transporte  
0    Razonablemente Buena            1104.0          702.0            66.0  
1    Razonablemente Buena            1104.0          702.0            66.0  
2    Razonablemente Buena            1104.0          702.0            66.0  
3    Razonablemente Buena            1104.0          702.0            66.0  
4    Razonablemente Buena            1104.0          702.0            66.0  
..  

In [18]:
# CHECKING HOW MANY NULL VALUES ARE IN MY DF
barrios_gdf.isnull().sum()

geometry                  0
coddistrit                0
gis_gis_barrios_area     54
object_id_barrio          0
linkid                    0
codbarrio                 0
coddistbar                0
geo_point_2d              0
nombre_barrio             0
%_zona_verde             27
nivel_acustico            0
num_hospitales           62
num_colegios             18
num_chargestations      100
calidad_ambiental        78
num_contenedores          6
num_papeleras             6
num_transporte            8
dtype: int64